lexicon/providers/mythicbeasts.py

```diff
@@ -0,0 +1,280 @@
+import time as time
+
+import requests
+
+from lexicon.providers.base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ["mythic-beasts.com"]
+
+
+def provider_parser(subparser):
+    """Return the parser for this provider"""
+    subparser.description = """
+        There are two ways to provide an authentication granting access to the Mythic Beasts API
+        1 - With your API credentials (user/password),
+            with --auth-username and --auth-password flags.
+        2 - With an API token, using --auth-token flags.
+        These credentials and tokens must be generated using the Mythic Beasts API v2.
+    """
+    subparser.add_argument(
+        "--auth-username",
+        help="specify API credentials username",
+    )
+    subparser.add_argument(
+        "--auth-password",
+        help="specify API credentials password",
+    )
+    subparser.add_argument(
+        "--auth-token",
+        help="specify API token for authentication",
+    )
+
+
+class Provider(BaseProvider):
+    """Provider class for Mythic Beasts"""
+
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = "https://api.mythic-beasts.com/dns/v2"
+        self.auth_token = None
+
+    def _authenticate(self):
+        payload = self._get("/zones")
+
+        if self.domain is None:
+            if not payload["zones"]:
+                raise Exception("No domain found")
+            if len(payload["zones"]) > 1:
+                raise Exception("Too many domains found. This should not happen")
+            else:
+                self.domain = payload["zones"][0]
+        else:
+            if not payload["zones"]:
+                raise Exception("No domain found")
+            if self.domain not in payload["zones"]:
+                raise Exception("Requested domain not found")
+
+        self.domain_id = hashlib.md5(self.domain.encode("utf-8")).hexdigest()
+
+    # Create record. If record already exists with the same content, do nothing'
+    def _create_record(self, rtype, name, content):
+        LOGGER.debug("type %s", rtype)
+        LOGGER.debug("name %s", name)
+        LOGGER.debug("content %s", content)
+
+        if rtype == 'CNAME':
+            content = self._fqdn_name(content)
+
+        data = {
+            "records": [
+                {
+                    "host": self._relative_name(name),
+                    "type": rtype,
+                    "data": content,
+                }
+            ]
+        }
+        if self._get_lexicon_option("ttl"):
+            data["records"][0]["ttl"] = self._get_lexicon_option("ttl")
+
+        payload = {"success": True}
+        try:
+            payload = self._post(f"/zones/{self.domain}/records", data)
+        except requests.exceptions.HTTPError as err:
+            if err.response.status_code != 400 and err.response.json()["errors"][0][0:16] != 'Duplicate record':
```

Comment:
Maybe you could add a log debug entry here to say that the record already exists.